core/src/main/java/org/apache/iceberg/view/ViewHistoryEntryParser.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+class ViewHistoryEntryParser {
+
+  private ViewHistoryEntryParser() {}
+
+  static final String VERSION_ID = "version-id";
+  static final String TIMESTAMP_MS = "timestamp-ms";
+
+  static String toJson(ViewHistoryEntry entry) {
+    return JsonUtil.generate(gen -> toJson(entry, gen), false);
+  }
+
+  static void toJson(ViewHistoryEntry entry, JsonGenerator generator) throws IOException {
+    generator.writeStartObject();
+    generator.writeNumberField(TIMESTAMP_MS, entry.timestampMillis());
+    generator.writeNumberField(VERSION_ID, entry.versionId());
+    generator.writeEndObject();
+  }
+
+  static ViewHistoryEntry fromJson(String json) {
+    Preconditions.checkArgument(
+        json != null && !json.isEmpty(),
+        "Cannot parse view history entry from invalid JSON: %s",
+        json);
+    return JsonUtil.parse(json, ViewHistoryEntryParser::fromJson);
+  }
+
+  static ViewHistoryEntry fromJson(JsonNode node) {
+    return new BaseViewHistoryEntry(
```

Comment:
I think this requires 2 precondition checks here:

```
Preconditions.checkArgument(null != json, "Cannot parse view history entry from null object");
Preconditions.checkArgument(json.isObject(), "Cannot parse view history entry from non-object: %s", json);
```

and then the `fromJson(String json)` method could be changed to 

```
static ViewHistoryEntry fromJson(String json) {
    return JsonUtil.parse(json, ViewHistoryEntryParser::fromJson);
  }
```

Reply:
Separated the precondition checks.